### PR TITLE
[core] Fix UTIL worker issue

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
In PR #14594 GENERAL_IO_WORKER/RUNTIME_ENV_WORKER is not updated to the new name which will lead to py worker failure. This PR fix this.

Related PR: #14823 #14910

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
@rkooo567 can you help check it to make sure I didn't miss anything? Thanks!

**Comment by rkooo567** [comment]
`test_runtime_env` still seems to be flaky with this PR?

File: python/ray/worker.py [hunk 1]
```diff
@@ -1138,7 +1138,8 @@ def connect(node,
     worker.redis_client = node.create_redis_client()
 
     # Initialize some fields.
-    if mode in (WORKER_MODE, RESTORE_WORKER_MODE, SPILL_WORKER_MODE):
+    if mode in (WORKER_MODE, RESTORE_WORKER_MODE, SPILL_WORKER_MODE,
```
Comment (rkooo567) [inline python/ray/worker.py:1141]: Can you also add UTIL_WORKER_MODE to line 1245?

File: python/ray/_raylet.pyx [hunk 2]
```diff
@@ -748,6 +749,11 @@ cdef void delete_spilled_objects_handler(
                     ray_constants.WORKER_PROCESS_TYPE_RESTORE_WORKER_IDLE)
                 proctitle = (
                     ray_constants.WORKER_PROCESS_TYPE_RESTORE_WORKER_DELETE)
+            elif <int> worker_type == <int> WORKER_TYPE_UTIL_WORKER:
```
Comment (rkooo567) [inline python/ray/_raylet.pyx:752]: Why do we need this for delete_spilled_objects?

**Comment by iycheng** [comment]
> `test_runtime_env` still seems to be flaky with this PR?

It looks better. All green

File: python/ray/_raylet.pyx (hunk 2 above)
Comment (iycheng) [reply python/ray/_raylet.pyx:752]: Sorry, I guess I only searched the SPILL_WORKER and try to add it there, but not check them carefully. Thanks!

**Comment by iycheng** [comment]
@ericl This is the PR reverted which leads to runtime_env_test flaky. Should we consider it as a release blocker?

**Comment by ericl** [comment]
Yeah we should probably hotfix this, if it's breaking things in the 1.3
branch.

**Comment by iycheng** [comment]
Since runtime_env is not used in 1.3, it should make things broken, I think ...

**Comment by ericl** [comment]
Tests still look broken

**Comment by rkooo567** [comment]
@iycheng Isn't this a release blocker? This impacts the stability of object spilling right?

**Comment by iycheng** [comment]
If no one is using runtime_env then it's not a blocker. Since we plan to public release runtime env short after the 1.3 release, probably it's OK? @ericl  how do you think about this?

**Comment by ericl** [comment]
Yeah I don't think this is considered a released feature.

**Comment by rkooo567** [comment]
Oh, but I remember before you reverted the fix, there were some object spilling test failures that was invoked by UTIL worker. That doesn't exist anymore without merging this PR?

**Comment by rkooo567** [comment]
@ericl @iycheng 

I am seeing this error log when I ran streaming shuffle.

```
raylet) Traceback (most recent call last):                                                    | 0/1000 [02:29<?, ?it/s]
(raylet)   File "/home/ubuntu/ray/python/ray/workers/default_worker.py", line 133, in <module>
(raylet)     raise ValueError("Unknown worker type: " + args.worker_type)
(raylet) ValueError: Unknown worker type: UTIL_WORKER
```

So I think we should fix this for 1.3?

**Comment by iycheng** [comment]
> @ericl @iycheng
> 
> I am seeing this error log when I ran streaming shuffle.
> 
> ```
> raylet) Traceback (most recent call last):                                                    | 0/1000 [02:29<?, ?it/s]
> (raylet)   File "/home/ubuntu/ray/python/ray/workers/default_worker.py", line 133, in <module>
> (raylet)     raise ValueError("Unknown worker type: " + args.worker_type)
> (raylet) ValueError: Unknown worker type: UTIL_WORKER
> ```
> 
> So I think we should fix this for 1.3? (although this doesn't impact the correctness, this seems pretty bad to see this when running object spilling).

Got it. Let me see the effort of fixing the failing test cases.

**Comment by iycheng** [comment]
It looks like the test is fine.

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
@ericl I am seeing this a lot whenenver there's spilling. Any objection to pick this up to the 1.3 branch?

**Comment by rkooo567** [comment]
@iycheng Can you communicate with @amogkam to pick up this change?

**Comment by ericl** [comment]
RLlib ci failed due to npm network issue:
```

+++install_ray
+++cd /home/travis/build/ray-project/ray/ci/travis/../../python
+++build_dashboard_front_end
+++'[' linux-gnu = msys ']'
+++cd ray/new_dashboard/client
+++'[' -z '' ']'
+++set +x
npm ERR! code ERR_SOCKET_TIMEOUT
npm ERR! errno ERR_SOCKET_TIMEOUT
npm ERR! request to https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.10.4.tgz failed, reason: Socket timeout
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2021-04-01T01_58_26_145Z-debug.log
```

  /usr/bin/clang -fno-strict-aliasing -Wsign-compare -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch i386 -arch x86_64 -g -DNDEBUG=1 -I/Library/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c immutables/_map.c -o build/temp.macosx-10.6-intel-3.6/immutables/_map.o -O2 -std=c99 -fsigned-char -Wall -Wsign-compare -Wconversion
  clang: error: no such file or directory: 'immutables/_map.c'
  clang: error: no input files
  error: command '/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for immutables
ERROR: Could not build wheels for aiohttp, multidict, yarl which use PEP 517 and cannot be installed directly

Not sure about above, might be worth re-running.

**Comment by rkooo567** [comment]
I just re-ran both failed builds.

**Comment by iycheng** [comment]
> RLlib ci failed due to npm network issue:
> 
> ```
> 
> +++install_ray
> +++cd /home/travis/build/ray-project/ray/ci/travis/../../python
> +++build_dashboard_front_end
> +++'[' linux-gnu = msys ']'
> +++cd ray/new_dashboard/client
> +++'[' -z '' ']'
> +++set +x
> npm ERR! code ERR_SOCKET_TIMEOUT
> npm ERR! errno ERR_SOCKET_TIMEOUT
> npm ERR! request to https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.10.4.tgz failed, reason: Socket timeout
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/travis/.npm/_logs/2021-04-01T01_58_26_145Z-debug.log
> ```
> 
> ## /usr/bin/clang -fno-strict-aliasing -Wsign-compare -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch i386 -arch x86_64 -g -DNDEBUG=1 -I/Library/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c immutables/_map.c -o build/temp.macosx-10.6-intel-3.6/immutables/_map.o -O2 -std=c99 -fsigned-char -Wall -Wsign-compare -Wconversion
> clang: error: no such file or directory: 'immutables/_map.c'
> clang: error: no input files
> error: command '/usr/bin/clang' failed with exit status 1
> ERROR: Failed building wheel for immutables
> ERROR: Could not build wheels for aiohttp, multidict, yarl which use PEP 517 and cannot be installed directly
> 
> Not sure about above, might be worth re-running.

I didn't re-run it because I checked buildkite and it's good. So I felt it should be fine.

Let's wait for the re-run.

It looks like there is no mac test for buildkite?

**Comment by rkooo567** [comment]
There’s no Mac build in the buildkite

**Comment by ericl** [comment]
Hmm it failed again. Can you pull in master and retry?

**Comment by ericl** [comment]
Could it be related to the UNREACHABLE macro change?

**Comment by iycheng** [comment]
@ericl  RL test looks ok. The failed one is "xcode7 | MAC_WHEELS=1 MAC_JARS=1"
It timed out actually during the build, which from my previous experience, it's usually ok.
Do you think it's good to go or do you prefer a merge?

**Comment by ericl** [comment]
I'm pretty sure that build is stable in the flaky test dashboard. Given that it's failed multiple times here, isn't it more likely broken due to this pr?

**Comment by iycheng** [comment]
> I'm pretty sure that build is stable in the flaky test dashboard. Given that it's failed multiple times here, isn't it more likely broken due to this pr?

It timeout, instead of failing I think. I can merge to master to give it another try. From my previous experience most time it's because that it just took too long to finish. When I merge cpp14 PR, it also timeout.

**Comment by ericl** [comment]
image

It seems like there was 1-2 suspicious timeouts, but it seems pretty rare. @simon-mo do you know if the PR builds have shorter timeouts than master?

**Comment by simon-mo** [comment]
Seems like all passed now? The timeout might happen when the pr builders can't hit the bazel cache. (only master build write to cache)

**Comment by ericl** [comment]
Could we raise the time limit for the PR builders so they can finish even if there isn't a cache hit?

**Pull request merged**